src/test/java/com/google/sps/UserServletTest.java

```diff
@@ -0,0 +1,157 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import com.google.gson.Gson;
+
+@RunWith(JUnit4.class)
+public final class UserServletTest {
+    private static final String ID = "userid";
+    private static final String EMAIL = "user@gmail.com";
+    private static final String NAME = "username";
+    private static final String SELF_INTRODUCTION = "I am the user";
+    private static final String IMG_URL = "/img.com";
+
+    private static final User toSaveUser = new User.Builder(ID, EMAIL).setName(NAME).addSelfIntroduction(SELF_INTRODUCTION).addImgUrl(IMG_URL).build();
+    private static User toGetUser; 
+   
+    private UserServlet userServlet;
+    private Map<String, Object> attributeToValue = new HashMap<>();
+    private LocalServiceTestHelper helper;
+
+    @Before
+    public void setup() {
+        // Set the userdata that the Userservice will return. 
+        attributeToValue.put("com.google.appengine.api.users.UserService.user_id_key", (Object) ID); 
+        helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig())
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("localhost")
+          .setEnvEmail(EMAIL)
+          .setEnvAttributes(attributeToValue);
+        helper.setUp();
+
+        userServlet = new UserServlet();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+ 
+    @Test
+    public void doPostDoGet_existingUserWithAllDataSet_returnsUserWithAllData() throws IOException, ServletException {
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter(UserServlet.NAME_INPUT)).thenReturn(NAME);
+        when(request.getParameter(UserServlet.SELF_INTRODUCTION_INPUT)).thenReturn(SELF_INTRODUCTION);
+        when(request.getParameter(UserServlet.PUBLIC_PORTFOLIO_INPUT)).thenReturn(UserServlet.PUBLIC_PORTFOLIO_INPUT_PUBLIC_VALUE);
+
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+         
+        when(response.getWriter()).thenReturn(pw);
+
+        // Save the currenly logged in user's data
+        userServlet.doPost(request, response);
+        // Get the currenly logged in user's previously saved data
+        userServlet.doGet(request, response);
+
+        pw.flush();
+        Gson gson = new Gson();
+        User resultUser = gson.fromJson(sw.toString(), User.class);
+        assertEquals(ID, resultUser.getId());
+        assertEquals(EMAIL, resultUser.getEmail());
+        assertEquals(NAME, resultUser.getName());
+        assertTrue(resultUser.portfolioIsPublic());
+        assertEquals(SELF_INTRODUCTION, resultUser.getSelfIntroduction());
+    }
+
+    @Test
+    public void doPostDoGet_existingUserWithNoDataSet_returnsUserWithMissingData() throws IOException, ServletException {
```

Comment:
Split this method to 2 different methods, one for doPost and one for doGet.

Reply:
Done.